### PR TITLE
feat(data-grid): update styles to align with new Paste Twilio theme

### PR DESCRIPTION
**Pull request by SiTaggart**
Updates to the DataGrid to follow Table styles where we have some custom implementations of Table elements inside DataGrid.

## Checklist

- [ ] Work in a "Draft" branch whilst you are getting feedback to reduce Chromatic costs. Once you are ready for approval, convert the PR to "Ready to review"
- [ ] Ensure all `required` checks have passed
- [ ] Icon changes must have a product design review
- [ ] Website content changes must have a product design review
- [ ] At least two engineers have reviewed any code changes
- [ ] At least one engineering lead has reviewed any core package changes or repository infrastructure
- [ ] Website visual regression testing is run by adding `🕵🏻‍♀️ Run website visual regression` label

### CODE REVIEW

**Review by nkrantz — COMMENTED** [review]
code lgtm, I'll approve when chromatic things are fixed and you get design approval

**Review by nkrantz — APPROVED** [review]

**Review by loreina — DISMISSED** [review]

**Pull request merged**
